packages/vscode-extension/src/webview/components/UrlSelect.tsx

```diff
@@ -22,59 +60,208 @@ interface UrlSelectProps {
+                e.preventDefault();
+                closeDropdownWithValue(textfieldRef.current?.value ?? "");
+                textfieldRef.current?.blur();
+              }
+              if (e.key === "Escape") {
+                setIsDropdownOpen(false);
+                textfieldRef.current?.blur();
+              }
+              if (e.key === "ArrowDown") {
+                if (isDropdownOpen) {
+                  navigateBetweenItems(
+                    e,
+                    undefined,
+                    document.querySelector(".url-select-item") as HTMLDivElement
+                  );
+                } else {
+                  setIsDropdownOpen(true);
+                }
+              }
+            }}
+          />
+        </Popover.Trigger>
+
+        <Popover.Content
+          className="url-select-content"
+          side="bottom"
+          onEscapeKeyDown={() => setIsDropdownOpen(false)}
+          onPointerDownOutside={(e) => {
+            const input = textfieldRef.current;
+            const originalEvent = e.detail.originalEvent as PointerEvent;
+            const elemRect = (e.target as HTMLDivElement).getBoundingClientRect();
+            if (
+              !input ||
+              originalEvent.clientX <= elemRect.left ||
+              originalEvent.clientX >= elemRect.right ||
+              originalEvent.clientY <= elemRect.top ||
+              originalEvent.clientY >= elemRect.bottom
+            ) {
+              setIsDropdownOpen(false);
+            }
+          }}>
+          <div className="url-select-viewport">
+            {(filteredItems.length > 0 || filteredOutItems.length > 0) && (
+              <div className="url-select-separator no-top-margin" />
+            )}
+
+            {filteredItems.length > 0 && (
+              <div className="url-select-group url-select-group-suggested">
+                <div className="url-select-label">Suggested paths:</div>
+                {filteredItems.map(
+                  (item) =>
+                    item.name && (
+                      <PopoverItem
+                        item={item}
+                        width={textfieldWidth}
+                        onClose={closeDropdownWithValue}
+                        onNavigate={navigateBetweenItems}
+                        getNameFromId={getNameFromId}
+                        textfieldRef={textfieldRef as React.RefObject<HTMLInputElement>}
+                      />
+                    )
+                )}
+              </div>
+            )}
+
+            {filteredItems.length > 0 && filteredOutItems.length > 0 && (
+              <div className="url-select-separator" />
+            )}
+
+            {filteredOutItems.length > 0 && (
+              <div className="url-select-group url-select-group-other">
+                <div className="url-select-label">Other paths:</div>
+                {filteredOutItems.map(
+                  (item) =>
+                    item.name && (
```

Comment:
NIT: You could probably filter out `item`s with `item.name === ""` in the effect where you set `filteredItems` and `filteredOutItems`, but the current approach is also good (since you can see that only non-empty items are rendered). Feel free to do whichever you think is better.